lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -435,13 +533,36 @@ def apply(self, resource_reqs: ResourceRequirements) -> None:
             field_manager=self.__class__.__name__,
         )
 
+    def dry_run_apply(self, resource_reqs: ResourceRequirements):
+        """Run a dry-run patch operation."""
+        # Read the token for authentication
+        with open(self.token_path, "r") as token_file:
+            token = token_file.read()
+
+        patch_url = f"{self.api_url}/apis/apps/v1/namespaces/{self.namespace}/statefulsets/{self.statefulset_name}"
+
+        headers = {
+            "Authorization": f"Bearer {token}",
+            "Content-Type": "application/apply-patch+yaml",
+            "Accept": "application/json",
+        }
+
+        with httpx.Client(verify=self.ca_path) as client:
```

Comment:
Why do we need httpx?
I.e. we were able to use lighkube.Client before, can we continue using lightkube instead of httpx directly?
(Seems like lightkube is able to figure out api url and token from the env on its own.)

If lightkube doesn't support the dryRun param yet, would it be straightforward to contribute the feature there?

Reply:
lightkube's client doesn't have the support yet.
I opened an enhancement proposal https://github.com/gtsystem/lightkube/issues/73, which I can contribute to adding the support needed. 
But for the purpose of this PR, I think we can continue using httpx and add a TODO?

Reply:
Merged in [v0.15.4](https://github.com/gtsystem/lightkube/releases/tag/v0.15.4)